src/button/AsyncButton.tsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from "react";
+
+import { Spinner } from "../common/Spinner";
+import { Button } from "./Button";
+
+import type { FC, ReactNode } from "react";
+import type { ButtonProps as AriaButtonProps } from "react-aria-components";
+
+import type { ButtonVariantProps } from "../../styled-system/recipes";
+import type { WithoutClassName } from "../types";
+
+export interface AsyncButtonProps extends Omit<WithoutClassName<AriaButtonProps>, "onPress">, Partial<ButtonVariantProps> {
+  children?: ReactNode | ((renderProps: { isPending: boolean }) => ReactNode);
+  loadingText?: string;
+  onPress?: (e: Parameters<NonNullable<AriaButtonProps["onPress"]>>[0]) => void | Promise<void>;
+}
+
+/**
+ * Async button component that extends Button with automatic loading state handling.
```

Comment:
Lets clean up this comment, from a practical standpoint why does this Button exist and how can it be used?